crates/nargo_cli/src/resolver.rs

```diff
@@ -85,181 +37,34 @@ struct CachedDep {
 /// Note that the backend is ignored in the dependencies.
 /// Since Noir is backend agnostic, this is okay to do.
 /// XXX: Need to handle when a local package changes!
-pub(crate) fn resolve_root_manifest(
-    dir_path: &std::path::Path,
-    package: Option<String>,
-) -> Result<(Context, CrateId), DependencyResolutionError> {
-    let fm = FileManager::new(dir_path);
-    let graph = CrateGraph::default();
-    let mut context = Context::new(fm, graph);
 
-    let manifest_path = super::find_package_manifest(dir_path)?;
-    let manifest = super::manifest::parse(&manifest_path)?;
-
-    let crate_id = match manifest {
-        Manifest::Package(package) => {
-            let (entry_path, crate_type) = super::lib_or_bin(dir_path)?;
-
-            let crate_id = create_local_crate(&mut context, &entry_path, crate_type);
-            let pkg_root = manifest_path.parent().expect("Every manifest path has a parent.");
-
-            resolve_package_manifest(&mut context, crate_id, package, pkg_root)?;
-
-            crate_id
-        }
-        Manifest::Workspace(workspace) => resolve_workspace_manifest(
-            &mut context,
-            package,
-            manifest_path,
-            dir_path,
-            workspace.config,
-        )?,
-    };
-
-    Ok((context, crate_id))
-}
-
-// Resolves a config file by recursively resolving the dependencies in the config
-// Need to solve the case of a project trying to use itself as a dep
-//
-// We do not need to add stdlib, as it's implicitly
-// imported. However, it may be helpful to have the stdlib imported by the
-// package manager.
-fn resolve_package_manifest(
+pub(crate) fn prepare_dependencies(
```

Comment:
`resolver` is just a holdover from the code that was already here. There's definitely a better place for these functions, but it hasn't surfaced itself yet.